server/migrations/20240327015248-add-user-role.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    await queryInterface.addColumn("users", "role", {
+      type: Sequelize.ENUM("admin", "member", "viewer", "guest"),
+      allowNull: true,
+    });
+    if (process.env.DEPLOYMENT === "hosted") {
+      return;
+    }
+
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'viewer' WHERE "isViewer" = true`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
+      }
+    );
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'admin' WHERE "isAdmin" = true`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
+      }
+    );
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'member' WHERE role IS NULL`,
```

Comment:
Can impose a non-null constraint on `role` through a follow up query.

Reply:
Will add in later PR to allow for async backfill in cloud